### PR TITLE
Implementation of EPG Model for the media player entity

### PR DESCRIPTION
**Pull request by miloit**
Implementation of EPG Model for the media player entity

### CODE REVIEW

File: components/media_player/ui/CardTVChannelList [hunk 1]
```diff
@@ -0,0 +1,253 @@
+/******************************************************************************
+ *
+ * Copyright (C) 2018-2019 Marton Borzak <hello@martonborzak.com>
```
Comment (zehnm) [inline components/media_player/ui/CardTVChannelList]: Is this Marton's clone?

File: dependencies.cfg [hunk 2]
```diff
@@ -4,3 +4,4 @@
 #   separator: colon + optional white space
 #   version: git branch or tag
 integrations.library: v0.6.0
```
Comment (zehnm) [inline dependencies.cfg]: I've just release v0.7.0
Please update and test if there's no conflict.

File: remote.pro [hunk 3]
```diff
@@ -506,6 +506,7 @@ win32 {
 
 DISTFILES += \
     components/FpsItem.qml \
+    components/media_player/ui/EPGItemView.qml \
```
Comment (zehnm) [inline remote.pro]: Does this belong in here? I'm still not very experienced with qml, so let's ask @martonborzak 
My FpsItem.qml might be wrong here as well...

File: components/media_player/ui/CardMediaPlayerEPGView.qml [hunk 4]
```diff
@@ -36,43 +37,45 @@ Rectangle {
...
+        //if (start){
+        console.debug("LOAD USER EPG");
+        //  start = false;
+        obj.getMediaPlayerEPGView("all");
+        obj.browseModelChanged.connect(onFirstLoadComplete);
+        //}
+        /*if (!isCurrentItem) {
             tvChannelListListView.contentY = 0-120;
-        }
+        }*/
```
Comment (zehnm) [inline components/media_player/ui/CardMediaPlayerEPGView.qml]: What's with the start flag?

File: components/media_player/ui/CardMediaPlayerEPGView.qml [hunk 5]
```diff
@@ -22,6 +22,7 @@
 import QtQuick 2.12
 import QtQuick.Controls 2.12
 import Style 1.0
+//import QtQml.Models 2.2
```
Comment (zehnm) [inline components/media_player/ui/CardMediaPlayerEPGView.qml]: If not used, please remove

**Review by zehnm — CHANGES_REQUESTED** [review]
Just a few cosmetic things. Otherwise looks good to me.
For the qml functionality I can't say much though.

File: components/media_player/ui/CardTVChannelList (hunk 1 above)
Comment (miloit) [reply components/media_player/ui/CardTVChannelList]: this is not even a qml file.....It's always there but Idelet it again

File: dependencies.cfg (hunk 2 above)
Comment (miloit) [reply dependencies.cfg]: done

File: remote.pro (hunk 3 above)
Comment (miloit) [reply remote.pro]: solved for me

File: components/media_player/ui/CardMediaPlayerEPGView.qml (hunk 4 above)
Comment (miloit) [reply components/media_player/ui/CardMediaPlayerEPGView.qml]: forgot to uncomment

File: components/media_player/ui/CardMediaPlayerEPGView.qml (hunk 5 above)
Comment (miloit) [reply components/media_player/ui/CardMediaPlayerEPGView.qml]: removed

**Review by zehnm — APPROVED** [review]

File: components/media_player/ui/CardMediaPlayerEPGView.qml [hunk 6]
```diff
@@ -90,129 +93,85 @@ Rectangle {
     SwipeView {
         id: swipeView
         width: parent.width; height: parent.height
+        opacity: 0.9
```
Comment (martonborzak) [inline components/media_player/ui/CardMediaPlayerEPGView.qml]: Why the opacity?

File: components/media_player/ui/CardMediaPlayerEPGView.qml [hunk 7]
```diff
@@ -90,129 +93,85 @@ Rectangle {
...
-                            source: item_image
-                            anchors.fill: parent
-                            fillMode: Image.PreserveAspectCrop
-                            asynchronous: true
+                        id: epgItemRectangle
+                        x: 20 + item_xCoordinate
+                        y: 10 + item_column * item_height
+                        height: item_height
+                        width: item_width
+                        border.color: "#000000"
```
Comment (martonborzak) [inline components/media_player/ui/CardMediaPlayerEPGView.qml]: Please use colors from the Style singleton.

File: components/media_player/ui/CardMediaPlayerEPGView.qml [hunk 8]
```diff
@@ -90,129 +93,85 @@ Rectangle {
...
+                        color: item_color
+                        Text {
+                            id: epgItemTitleText
+                            text: item_title
+                            elide: Text.ElideRight
+                            width: epgItemRectangle.width
+                            wrapMode: Text.NoWrap
+                            color: "#000000"
```
Comment (martonborzak) [inline components/media_player/ui/CardMediaPlayerEPGView.qml]: Please use colors from the Style singleton.

**Review by martonborzak — CHANGES_REQUESTED** [review]

File: components/media_player/ui/CardMediaPlayerEPGView.qml (hunk 8 above)
Comment (miloit) [reply components/media_player/ui/CardMediaPlayerEPGView.qml]: Solved

File: components/media_player/ui/CardMediaPlayerEPGView.qml (hunk 7 above)
Comment (miloit) [reply components/media_player/ui/CardMediaPlayerEPGView.qml]: Resolved

File: components/media_player/ui/CardMediaPlayerEPGView.qml (hunk 6 above)
Comment (miloit) [reply components/media_player/ui/CardMediaPlayerEPGView.qml]: Deleted

File: components/media_player/ui/CardMediaPlayerEPGView.qml [hunk 9]
```diff
@@ -36,19 +36,22 @@ Rectangle {
...
-        if (!isCurrentItem) {
-            tvChannelListListView.contentY = 0-120;
+            if (start){
+                console.debug("LOAD USER EPG");
+                start = false;
+                obj.getMediaPlayerEPGView("all");
+                obj.browseModelChanged.connect(onFirstLoadComplete);
+            }
+            if (!isCurrentItem) {
+                epgItemListListView.contentY = 0-120;
```
Comment (martonborzak) [inline components/media_player/ui/CardMediaPlayerEPGView.qml:53]: Could be just `-120` :)

**Review by martonborzak — APPROVED** [review]

**Pull request merged**